shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -61,6 +61,14 @@
  *   <li>{@link #getDartEntrypointFunctionName()}</li>
  *   <li>{@link #getInitialRoute()}</li>
  * </ul>
+ * <p>
+ * {@code FlutterActivity} can be used with a cached {@link FlutterEngine} instead of creating a new
+ * one. Use {@link #withCachedEngine(String)} to build a {@code FlutterActivity} {@code Intent} that
```

Comment:
Should this be accompanied with some documentation change on when to use a cached engine?

Reply:
Is there some existing documentation that you think should be updated?

Reply:
I am not sure where the upstream user documentation for the new embedding is. Internally, it would be as part of the FAQ.

> FlutterActivity *can* be used with a cached FlutterEngine.

but why and when should I use it with a cached Flutter Engine?

Reply:
@mehmetf @xster the only 2 times that I can think of that you wouldn't want to warm up an engine is:

1. FlutterActivity is the first Activity that is launched, therefore rendering warm up redundant, or

2. You have no idea when/if you'll need a Flutter experience and you don't want to waste the resources.

Can either of you think of other cases?

Reply:
I think that's right. We can probably phrase the doc to say using a cached engine is generally recommended for optimal launch latency except where impossible such as in the cases listed.

Reply:
Updated docs to reflect this.